app/models/vm_or_template/operations.rb

```diff
@@ -71,30 +71,21 @@ def vm_destroy
   #
 
   def validate_vm_control_shelve_action
-    msg = validate_vm_control
-    return {:available => msg[0], :message => msg[1]} unless msg.nil?
+    unless supports_control?
+      return {:available => false, :message => unsupported_reason(:control)}
+    end
     return {:available => true,   :message => nil}  if %w(on off suspended paused).include?(current_state)
     {:available => false,  :message => "The VM can't be shelved, current state has to be powered on, off, suspended or paused"}
   end
 
   def validate_vm_control_shelve_offload_action
-    msg = validate_vm_control
-    return {:available => msg[0], :message => msg[1]} unless msg.nil?
+    unless supports_control?
+      return {:available => false, :message => unsupported_reason(:control)}
+    end
     return {:available => true,   :message => nil}  if %w(shelved).include?(current_state)
     {:available => false,  :message => "The VM can't be shelved offload, current state has to be shelved"}
   end
 
-  def validate_vm_control
-    # Check the basic require to interact with a VM.
-    return [false, 'The VM is retired'] if self.retired?
-    return [false, 'The VM is a template'] if self.template?
```

Comment:
As a reminder, I think `false` would indicate that the feature is generally unsupported by the class. Whereas `true` is: as a class I support it, but there are conditions that prevent me.